### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by serucee**

### CODE REVIEW

File: src/Model/Configuration.php [hunk 1]
```diff
@@ -31,36 +31,59 @@ class Configuration
...
 
     /**
-     * Fetch the given key from the loaded configuration
+     * Set configuration property
      *
-     * @param string|int $key
-     * @param bool $mandatory Throw error if value does not exist
+     * @since 0.0.1
+     */
+    protected function setConfiguration()
```
Comment (rinnhofer) [inline src/Model/Configuration.php]: maybe more context - we parse a file within this setter... or setConfiguration should be called with configuration content to ensure single responsibility?

File: src/Helper/FileParserAbstract.php [hunk 2]
```diff
@@ -22,12 +22,37 @@ abstract class FileParserAbstract implements Parser
...
+    }
+
+    /**
+     * @param $file
+     *
+     * @throws MissingFileException
+     *
+     * @since 0.0.1
+     */
+    protected function setFile($file)
```
Comment (rinnhofer) [inline src/Helper/FileParserAbstract.php]: do we need a setter for the file anyway? (YAGNI)

File: src/Helper/FileParserJson.php [hunk 3]
```diff
@@ -24,7 +24,7 @@ class FileParserJson extends FileParserAbstract
      */
     public function parse()
     {
-        $fileContentRaw = file_get_contents($this->file);
+        $fileContentRaw = file_get_contents($this->getFile());
```
Comment (rinnhofer) [inline src/Helper/FileParserJson.php]: do we want to be dependent on the logic from getFile?

File: src/Model/Configuration.php [hunk 4]
```diff
@@ -31,36 +31,59 @@ class Configuration
...
     /**
      * Configuration constructor.
      *
      * @param Parser $parser
      *
+     * @since 0.0.1
+     */
+    public function __construct(Parser $parser)
+    {
+       $this->setParser($parser);
```
[the remark this answers is not in the document]
Comment (serucee) [reply src/Model/Configuration.php]: Since setter is protected and only used at this place it does not really matter - this was suggested during a review.

**Review by rinnhofer — APPROVED** [review]

**Pull request merged**